src/internal/packager2/deploy.go

```diff
@@ -0,0 +1,704 @@
+			deployedComponent.InstalledCharts = installedCharts
+		}
+
+		deployedComponents = append(deployedComponents, deployedComponent)
+		idx := len(deployedComponents) - 1
+		if d.isConnectedToCluster() {
+			if _, err := d.c.RecordPackageDeployment(ctx, pkgLayout.Pkg, deployedComponents, packageGeneration); err != nil {
+				l.Debug("unable to record package deployment", "component", component.Name, "error", err.Error())
+			}
+		}
+		var charts []types.InstalledChart
+		var deployErr error
+		if pkgLayout.Pkg.IsInitConfig() {
+			charts, deployErr = d.deployInitComponent(ctx, pkgLayout, component, opts)
+		} else {
+			charts, deployErr = d.deployComponent(ctx, pkgLayout, component, false, false, opts)
+		}
+
+		onDeploy := component.Actions.OnDeploy
+
+		onFailure := func() {
+			if err := actions.Run(ctx, onDeploy.Defaults, onDeploy.OnFailure, d.vc); err != nil {
+				l.Debug("unable to run component failure action", "error", err.Error())
+			}
+		}
+
+		if deployErr != nil {
+			onFailure()
+			deployedComponents[idx].Status = types.ComponentStatusFailed
+			if d.isConnectedToCluster() {
+				if _, err := d.c.RecordPackageDeployment(ctx, pkgLayout.Pkg, deployedComponents, packageGeneration); err != nil {
+					l.Debug("unable to record package deployment", "component", component.Name, "error", err.Error())
+				}
+			}
+			return nil, fmt.Errorf("unable to deploy component %q: %w", component.Name, deployErr)
+		}
+
+		// Update the package secret to indicate that we successfully deployed this component
+		deployedComponents[idx].InstalledCharts = charts
+		deployedComponents[idx].Status = types.ComponentStatusSucceeded
+		if d.isConnectedToCluster() {
+			if _, err := d.c.RecordPackageDeployment(ctx, pkgLayout.Pkg, deployedComponents, packageGeneration); err != nil {
+				l.Debug("unable to record package deployment", "component", component.Name, "error", err.Error())
+			}
+		}
+
+		if err := actions.Run(ctx, onDeploy.Defaults, onDeploy.OnSuccess, d.vc); err != nil {
+			onFailure()
+			return nil, fmt.Errorf("unable to run component success action: %w", err)
+		}
+	}
+
+	return deployedComponents, nil
+}
+
+func (d *deployer) deployInitComponent(ctx context.Context, pkgLayout *layout.PackageLayout, component v1alpha1.ZarfComponent, opts DeployOpts) ([]types.InstalledChart, error) {
+	l := logger.From(ctx)
+	hasExternalRegistry := opts.RegistryInfo.Address != ""
+	isSeedRegistry := component.Name == "zarf-seed-registry"
+	isRegistry := component.Name == "zarf-registry"
+	isInjector := component.Name == "zarf-injector"
+	isAgent := component.Name == "zarf-agent"
+
+	// Always init the state before the first component that requires the cluster (on most deployments, the zarf-seed-registry)
+	if component.RequiresCluster() && d.s == nil {
+		applianceMode := false
+		for _, component := range pkgLayout.Pkg.Components {
+			if component.Name == "k3s" {
+				applianceMode = true
+			}
+		}
+		err := d.c.InitState(ctx, cluster.InitStateOptions{
```

Comment:
Noticing that `StorageClass` is not utilized here and there is no option in `DeployOpts` in order to propagate use of the `--storage-class` init flag. Is this intentional?